lambda_function.py

```diff
@@ -80,16 +80,35 @@ def elasticsearch_handler(event, context):
     print(f"Successfully processed {success}/{total} items for opensearch")
 
 
+def _send_to_splunk(events, errors, splunk_hec_url, splunk_hec_token):
+    try:
+        response = requests.post(
+            splunk_hec_url,
+            headers={'Authorization': f'Splunk {splunk_hec_token}'},
+            json=events,
+            timeout=30
+        )
+        response.raise_for_status()
+        return len(events)
+    except Exception as e:
+        errors.append(str(e))
+        return 0
+
 def splunk_handler(event, context):
-    secret = get_secret('splunk-quayio-audit-log')
+    secret = get_secret(os.environ["secret_name"])
     # splunk HEC configuration
     splunk_hec_url = secret["splunk_hec_url"]
     splunk_hec_token = secret["splunk_hec_token"]
     splunk_index = secret["splunk_index"]
+    success_count = 0
+    errors = []
     events = []
+    max_batch_size = 500
 
     for record in event["Records"]:
-        message, message_date = process_kinesis_record(record)
+        message, message_date = _process_kinesis_record(record)
```

Comment:
Can you do a pre-process for all Records, so we don't need to run this twice?

Reply:
good idea!